headerfs/file.go

```diff
@@ -49,50 +51,31 @@ func (h *headerStore) readRaw(seekDist uint64) ([]byte, error) {
 	return rawHeader[:], nil
 }
 
-// readHeaderRange will attempt to fetch a series of headers within the target
-// height range. This method batches a set of reads into a single system call
-// thereby increasing performance when reading a set of contiguous headers.
+// readHeaderRange will attempt to fetch a series of block headers within the
+// target height range. This method batches a set of reads into a single system
+// call thereby increasing performance when reading a set of contiguous
+// headers.
 //
 // NOTE: The end height is _inclusive_ so we'll fetch all headers from the
 // startHeight up to the end height, including the final header.
 func (h *blockHeaderStore) readHeaderRange(startHeight uint32,
 	endHeight uint32) ([]wire.BlockHeader, error) {
 
 	// Based on the defined header type, we'll determine the number of
-	// bytes that we need to read past the sync point.
-	var headerSize uint32
-	switch h.indexType {
-	case Block:
-		headerSize = 80
-
-	case RegularFilter:
-		headerSize = 32
-
-	default:
-		return nil, fmt.Errorf("unknown index type: %v", h.indexType)
-	}
-
-	// Each header is 80 bytes, so using this information, we'll seek a
-	// distance to cover that height based on the size of block headers.
-	seekDistance := uint64(startHeight) * uint64(headerSize)
-
-	// Based on the number of headers in the range, we'll allocate a single
-	// slice that's able to hold the entire range of headers.
-	numHeaders := endHeight - startHeight + 1
-	rawHeaderBytes := make([]byte, headerSize*numHeaders)
-
-	// Now that we have our slice allocated, we'll read out the entire
-	// range of headers with a single system call.
-	_, err := h.file.ReadAt(rawHeaderBytes, int64(seekDistance))
+	// bytes that we need to read from the file.
+	var headerSize uint32 = BlockHeaderSize
+	headerReader, err := readHeadersFromFile(
+		h.file, headerSize, startHeight, endHeight,
+	)
 	if err != nil {
 		return nil, err
 	}
 
 	// We'll now incrementally parse out the set of individual headers from
 	// our set of serialized contiguous raw headers.
-	headerReader := bytes.NewReader(rawHeaderBytes)
+	numHeaders := endHeight - startHeight + 1
 	headers := make([]wire.BlockHeader, 0, numHeaders)
-	for headerReader.Len() != 0 {
+	for i := uint32(0); i < numHeaders; i++ {
```

Comment:
Why this over the prior logic? Assuming all contents are written to disk in 80 byte chunks the prior works as well assuming the method still returns a ` bytes.Reader`.

Reply:
Reverted to returning `bytes.Reader`: https://github.com/lightninglabs/neutrino/pull/134/commits/324253acdbd2305da9bbda5b5de133bc8e480cf5